test/core/providers/local-accounts.ts

```diff
@@ -0,0 +1,131 @@
+import { assert } from "chai";
+import Transaction from "ethereumjs-tx";
+import { EventEmitter } from "events";
+
+import { getConfig } from "../../../src/core/config";
+import { IEthereumProvider } from "../../../src/core/providers/ethereum";
+import {
+  createLocalAccountsProvider,
+  hashTransaction,
+  signTransaction
+} from "../../../src/core/providers/local-accounts";
+import { expectErrorAsync } from "../../helpers/errors";
+import { useFixtureProject } from "../../helpers/project";
+
+class MockProvider extends EventEmitter implements IEthereumProvider {
+  public async send(method: string, params?: any[]): Promise<any> {
+    if (method === "eth_getTransactionCount") {
+      return 0x08;
+    }
+    if (method === "eth_sendRawTransaction") {
+      if (params !== undefined) {
+        return params[0];
+      }
+    }
+  }
+}
+
+describe("ethereum provider", () => {
+  let mock: MockProvider;
+  let wrapper: IEthereumProvider;
+
+  let accounts: string[] = [];
+  const chainId = 3;
+  useFixtureProject("config-project");
+
+  beforeEach(() => {
+    const [config, _] = getConfig();
+    accounts = config.networks.develop.accounts;
+    mock = new MockProvider();
+    wrapper = createLocalAccountsProvider(mock, accounts, chainId);
+  });
+
+  it("eth_accounts", async () => {
+    const response = await wrapper.send("eth_accounts");
+    // This should check that the address is derivated from the private key
+    assert.equal(response[0].length, 20);
+  });
+
+  it("eth_requestAccounts", async () => {
+    const response = await wrapper.send("eth_requestAccounts");
+    assert.equal(response[0].length, 20);
+  });
+
+  it("eth_sign", () => {
+    expectErrorAsync(
```

Comment:
You should use that helper in `async` functions and `await` it. This is a common catch using async/await, but once PR #103 is merged the linter will detect it.